datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -177,16 +177,16 @@ class except the :py:meth:`check` method but sometimes it might be useful for a
         }
 
         # Setup metric limits
-        try:
-            metric_limit = self.instances[0].get('max_returned_metrics', self.DEFAULT_METRIC_LIMIT)
+        if self.instance is not None:
```

Comment:
You can safely assume self.instance is set in agent 6. Legacy agent 5 has been branched out but we still have some legacy code on master.